src/Traits/QueueWithoutOverlap.php

```diff
@@ -31,12 +32,14 @@ public function queue($queue, $command)
     /**
      * Lock this job's key in redis, so no other
      * jobs can run with the same key.
+     * @param Lock|null $lock
      * @return bool - false if it fails to lock
      */
-    protected function acquireLock(array $lock = [])
+    protected function acquireLock(Lock $lock = null)
     {
         $lock_time = isset($this->lock_time) ? $this->lock_time : 300; // in seconds; 5 minutes default
-        $this->lock = RedLock::lock($lock['resource'] ?? $this->getLockKey(), $lock_time * 1000);
+        $resource = null !== $lock ? $lock->getResource() : $this->getLockKey();
```

Comment:
Also check against `$lock->getResource()` like it was before.
```suggestion
        $resource = null !== $lock ? $lock->getResource() ?? $this->getLockKey() : $this->getLockKey();
```